### ISSUE TITLE
Documentation: clarify hazelcast.wait.seconds.before.join

### ISSUE THREAD

**Issue by laszlocsomor** [hazelcast/hazelcast#16944]
**Describe the bug**

The [documentation](https://docs.hazelcast.org/docs/latest/manual/html-single/#system-properties) says too little about `hazelcast.wait.seconds.before.join` and `hazelcast.max.wait.seconds.before.join`.

**Expected behavior**

It'd be useful if the docs explained when to override these properties (and when not to), and gave guidance how to choose meaningful values. E.g. is the property relevant for only multicast discovery, or also for the other joiners? Is it safe to set to 0 for GCP / AWS / K8s?

It'd be equally reasonable to say that there was no general guidance (if it were the case), and that these properties should be set based on observation.

**To Reproduce**

n/a

**Additional context**

I tried Googling these properties to little avail. For example, [this StackOverflow answer]() seems informed but is the reasoning correct?

**Comment by mmedenjak** [comment]
@laszlocsomor thank you for the report. Would you be interested in sending a PR?

**Comment by laszlocsomor** [comment]
@mmedenjak : Certainly! Problem is, I'm looking for usage guidance myself :) If you can explain the semantics to me, and/or point me to the code where it's used so I can figure it out (`grep 'before.join'` gave no useful result), then I'm happy to write the text.

**Comment by mmedenjak** [comment]
I see `hazelcast.wait.seconds.before.join` is used in the `ClusterJoinManager` and `DiscoveryJoiner` and the `hazelcast.max.wait.seconds.before.join` property is used in `AbstractJoiner#getMaxJoinTimeToMasterNode`. Other than that, I don't remember the details and maybe it's best to first take a look at the code, understand what it's doing, and then submit the updated documentation, as that would make sure the documentation is up-to-date.

**Comment by laszlocsomor** [comment]
Thanks! I can take a look in the coming days. @Serdaro, do you intend to fix this or shall I send a PR?

**Comment by Serdaro** [comment]
Go ahead @laszlocsomor  :) 👍

**Comment by Serdaro** [comment]
@laszlocsomor , @mmedenjak , transferring this issue to the IMDG reference manual issue.
